### PR TITLE
Added pre-processed paths persistence through JSON files.

### PR DESCRIPTION
**Pull request by aalexandrasilva**
Paths can now be searched for and scored in a pre-processing step and persisted in a JSON file. 
Each property will produce a different file with all the corresponding paths. If the paths are expected to be pre-searched, but do not exist in file, we'll search for them. Same principle applies to scoring.

### CODE REVIEW

**Comment by aalexandrasilva** [comment]
Due to the incorrect resolution of the conflicts from `restructuring -> feature/preprocSearch`, the code was merged from `feature/preprocSearch` and `feature/pathsImport` (the branch `preprocSearch `was created from) to restructuring. Which means not all changes pertaining to this task are marked in this PR's changed files. 
Most differences that stem from the `pathsImport` branch were the way paths were dealt with once loaded into COPAAL, specifically esther's paths. These classes/interfaces can be found in `org.dice_research.fc.paths.imprt` package.

File: service/src/main/java/org/dice_research/fc/paths/imprt/PredicateRetriever.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+  /**
+   * Predicate variable in queries.
+   */
+  private final String PREDICATE_VARIABLE = "p";
+
+  /**
+   * Namespace filter to apply to queries.
+   */
+  private NamespaceFilter filter;
```
Comment (MichaelRoeder) [inline service/src/main/java/org/dice_research/fc/paths/imprt/PredicateRetriever.java:37]: I would suggest to have a `Predicate<String> filter`. This gives much more flexibility. You can of course provide a constructor which takes the namespace information and creates the predicate. However, I can imagine that we may want to exclude some specific properties, in the future :thinking: 

Another way to keep it easy might be to simply let NamespaceFilter extend the `Predicate<String>` interface and offer two constructors: one (easier) that takes a NamespaceFilter and a second (more flexible) constructor that takes any `Predicate<String>` implementation.

**Review by MichaelRoeder — COMMENTED** [review]
I only added a small comment. Apart from that, it looks very good :+1:

**Review by MichaelRoeder — APPROVED** [review]
LGTM

**Pull request merged**